addon/components/frost-expand.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Component definition for the frost-expand component
+ */
+import Ember from 'ember'
+const {observer} = Ember
+import layout from '../templates/components/frost-expand'
+import computed, {readOnly} from 'ember-computed-decorators'
+import Component from './frost-component'
+import {PropTypes} from 'ember-prop-types'
+
+export default Component.extend({
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+
+  layout,
+  tagName: 'div',
+  classNameBindings: ['cssStateClass'],
+
+  // == PropTypes =============================================================
+
+  /**
+   * Properties for this component. Options are expected to be (potentially)
+   * passed in to the component. State properties are *not* expected to be
+   * passed in/overwritten.
+   */
+  propTypes: {
+    // options
+    onExpand: PropTypes.func,
+    onCollapse: PropTypes.func,
+    expanded: PropTypes.bool,
+    duration: PropTypes.string,
+    expandLabel: PropTypes.string,
+    collapseLabel: PropTypes.string,
+    content: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.EmberComponent
+    ])
+    // state
+  },
+
+  /** @returns {Object} the default property values when not provided by consumer */
+  getDefaultProps () {
+    return {
+      // options
+      onExpand () {},
+      onCollapse () {},
+      expanded: false
+      // state
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  @readOnly
+  @computed('expanded')
+  /**
+   * Determine CSS state class name to use
+   * @param {Boolean} expanded - whether in expanded state or not
+   * @returns {String} the CSS class name to use
+   */
+  cssStateClass (expanded) {
+    return expanded ? 'expanded' : 'collapsed'
+  },
+
+  @readOnly
+  @computed('content')
+  /**
+   * Determine whether or not the content is a component
+   * @param {*} content - content to display
+   * @returns {Boolean} whether or not the content is a component
+   */
+  isComponentContent (content) {
+    return content !== null && typeof content !== 'string'
```

Comment:
We probably want to use the same type of validation that `prop-types` is using https://github.com/ciena-blueplanet/ember-prop-types/blob/master/addon/utils/validators/ember-component.js

In fact, if we could use that validator directly it would be safer